File: src/smartworkz-sdk-py/smartworkz/auth/authenticator.py
"""Authentication handling for SmartWorkz SDK."""

from typing import Dict
from smartworkz.config import SmartWorkzConfig


class Authenticator:
    """Handles authentication headers for API requests."""

    @staticmethod
    def get_headers(config: SmartWorkzConfig) -> Dict[str, str]:
        """
        Get authentication headers based on configuration.

        Args:
            config: SmartWorkzConfig instance

        Returns:
            Dictionary of headers including authentication

        Raises:
            ValueError: If configuration is invalid
        """
        config.validate()

        headers = {
            "Content-Type": "application/json",
            "User-Agent": "smartworkz-sdk-py/1.0.0",
        }

        if config.api_key:
            headers["X-API-Key"] = config.api_key
        elif config.bearer_token:
            headers["Authorization"] = f"Bearer {config.bearer_token}"

        return headers

File: src/smartworkz-sdk-js/src/auth/authenticator.ts
/**
 * Authentication handler for SmartWorkz SDK
 * Manages API key and Bearer token authentication
 */

import { SmartWorkzConfig } from '../config';

export class Authenticator {
  /**
   * Generate HTTP headers with authentication
   * @param config SDK configuration
   * @returns Object with authorization headers
   */
  static getHeaders(config: SmartWorkzConfig): Record<string, string> {
    const headers: Record<string, string> = {
      'Content-Type': 'application/json'
    };

    if (config.apiKey) {
      headers['X-API-Key'] = config.apiKey;
    } else if (config.bearerToken) {
      headers['Authorization'] = `Bearer ${config.bearerToken}`;
    }

    return headers;
  }
}
